src/app/components/validate-new-password/ValidateNewPassword.jsx

```diff
@@ -7,7 +7,8 @@ import notifications from "../../utilities/notifications";
 import ValidationItemList from "../../components/validation-item-list/ValidationItemList";
 
 const propTypes = {
-    updateValidity: PropTypes.func
+    updateValidity: PropTypes.func,
+    inputError: PropTypes.string
 };
 
 export class ValidateNewPassword extends Component {
```

Comment:
please simplify the state here. no need for complex object here for password. keep only dynamic values in the stae, remove the validation rules as they do not change.

Reply:
Also I would probably think about changing the name of the component NewPasswordInput.jsx.or just so in that case having state={type: 'password', password: '' .... }  would make sense .. or something similar. you can create seperate function to take password and pass it thru validation rules, if they don't pass show an error.

Reply:
I have cleaned up the state, however, we don't want to always show the error if they haven't met validation rules, only show it after a submit has happened (this is where the difference in the `passwordIsValid` and `showInputError` state vars come into play. I have added a gif to the PR showing how this works a little, sorry low quality (compression)